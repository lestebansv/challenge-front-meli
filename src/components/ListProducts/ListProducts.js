import { Link } from 'react-router-dom';
import CurrencyFormat from 'react-currency-format';
import styles from './ListProducts.module.scss';

/**
 * Component builds the product list search
 * 
 * @param {id, image, price, tittle, address, tags} param0 
 * @returns Render view product list
 */
const ListProducts = ({ id, image, price, tittle, address, tags }) => {
    return (
        <Link className={styles.card} to={{ pathname: `/items/${id}` }}>
            <div className={styles.products_card}>
                <img className={styles.img} src={image} alt={tags} />
            </div>

            <div>
                <p className={styles.price_shipping}>
                    <CurrencyFormat value={price} displayType={'text'} thousandSeparator={true} prefix={'$ '} />
                </p>
                <span className={styles.title_item}>
                    {tittle}
                </span>
            </div>

            <div className={styles.address}>
                <span>
                    {address}
                </span>
            </div>
        </Link>
    );
}

export default ListProducts;