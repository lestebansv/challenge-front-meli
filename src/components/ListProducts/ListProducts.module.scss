@import '../../assets/css/styles.scss';

body {
    background-color: $background-gray-light;
}

.address {
    color: $background-gray-heavy;
    font-size: 12px;
    margin-top: 11px;
    text-align: center;
}

.card {
    background-color: $background-white;
    color: black;
    display: grid;
    grid-template-columns: 196px 1fr 22%;
    min-width: 515px;
    padding: 1rem;
    position: relative;
    text-decoration: none;
}

.card:before {
    border-bottom: thin solid $background-gray-light;
    bottom: 0;
    content: "";
    height: 1px;
    left: 16px;
    position: absolute;
    width: 95%;
}

.card:first-child {
    border-radius: 4px 4px 0 0 !important;
}

.container {
    margin: 0 15%;
    text-align: justify;
}

.img {
    border-radius: 4px;
    max-height: 180px;
    max-width: 180px;
    vertical-align: middle;
}

.products_card {
    line-height: 180px;
    height: 180px;
    text-align: center;
    width: 180px;
}

p {
    margin-bottom: 32px !important;
    margin-top: 0 !important;
}

.price_shipping {
    font-size: 24px;
}

.title_item {
    font-size: 18px;
}