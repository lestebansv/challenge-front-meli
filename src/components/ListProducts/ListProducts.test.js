import { render } from '@testing-library/react';
import ListProducts from './ListProducts';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Link: ({ children, ...rest }) => <a data-testid='link' {...rest} >{children}</a>
}));

describe('ListProducts', () => {

    it('renders', () => {

        const id = 'MLA1135260991';
        const image = 'http://http2.mlstatic.com/D_674706-MLA40690247901_022020-I.jpg';
        const price = 35650;
        const tittle = 'E-reader  Kindle Paperwhite 10 Gen 8gb Azul Con Pantalla De 6  300ppp';
        const address = 'Capital Federal';
        const tags = [
            'good_quality_picture',
            'good_quality_thumbnail',
            'extended_warranty_eligible',
            'immediate_payment',
            'cart_eligible'
        ];

        const { container } = render(
            <ListProducts
                id={id}
                image={image}
                price={price}
                tittle={tittle}
                address={address}
                tags={tags}
            />
        );

        expect(container).toMatchSnapshot();
    });
    
});