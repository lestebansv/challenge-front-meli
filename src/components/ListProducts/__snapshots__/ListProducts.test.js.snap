// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListProducts renders 1`] = `
<div>
  <a
    class="card"
    data-testid="link"
    to="[object Object]"
  >
    <div
      class="products_card"
    >
      <img
        alt="good_quality_picture,good_quality_thumbnail,extended_warranty_eligible,immediate_payment,cart_eligible"
        class="img"
        src="http://http2.mlstatic.com/D_674706-MLA40690247901_022020-I.jpg"
      />
    </div>
    <div>
      <p
        class="price_shipping"
      >
        <span>
          $ 35,650
        </span>
      </p>
      <span
        class="title_item"
      >
        E-reader  Kindle Paperwhite 10 Gen 8gb Azul Con Pantalla De 6  300ppp
      </span>
    </div>
    <div
      class="address"
    >
      <span>
        Capital Federal
      </span>
    </div>
  </a>
</div>
`;
